### PR TITLE
Video Playback Fixes

### PR DESCRIPTION
**Pull request by mo123**
Fix some realmedia video files can't play.
~~Fix video playback from GoPro cameras.~~
Disable caching for mp4 files, they might have bad a/v interleaving and will cause cache thrashing.

## Types of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
jenkins build this please

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp [hunk 1]
```diff
@@ -1470,6 +1470,13 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
       {
         stream = new CDemuxStream();
         stream->type = STREAM_DATA;
+        // Video (mp4) from GoPro cameras can have a 'meta' track used for a file repair containing
+        //  'fdsc' data, this is also called the SOS track. Ignore it or we will glitch bad on startup.
+        if (pStream->codec->codec_tag == MKTAG('f','d','s','c'))
+        {
+          stream->SetDiscard(AVDISCARD_ALL);
+          pStream->discard = AVDISCARD_ALL;
+        }
```
Comment (a1rwulf) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: There's already a similar pull-request open:
https://github.com/xbmc/xbmc/pull/12579

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: check my comments on this PR

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamFile.cpp [hunk 2]
```diff
@@ -82,12 +82,17 @@ bool CDVDInputStreamFile::Open()
...
   std::string content = m_item.GetMimeType();
 
-  if (content == "video/mp4" ||
-      content == "video/x-msvideo" ||
-      content == "video/avi" ||
-      content == "video/x-matroska" ||
-      content == "video/x-matroska-3d")
+  if (content == "video/mp4")
+  {
+    flags |= READ_NO_CACHE;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamFile.cpp]: I don't think we want to disable caching in general for mp4

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp (hunk 1 above)
Comment (Rechi) [reply xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: Superceded by #13095

**Comment by Rechi** [comment]
@mo123 please share examples of realmedia videos and mp4s which aren't working correctly.

**Comment by mo123** [comment]
Hi

Most of the changes have been added by other PR's now.
I don't have RM test videos to share and the PR is probably not needed anymore.
I will open a new PR if there are still problems with RM videos in the future.

**Pull request closed without merging**